### PR TITLE
Fix NumberFormatException for EA Javas in some TCKs

### LINKED ISSUE #30227: Some TCKs use a version of awaitility that does not handle early access versions of Java
When running the latest MPFT and MPT TCKs with early access (pre-release) versions of Java, we hit the following stacktrace:
```
testAsyncDoubleCounter_EE11_FEATURES_MicroProfile_70:java.lang.NumberFormatException: For input string: "24-ea"
at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
at java.base/java.lang.Integer.parseInt(Integer.java:564)
at java.base/java.lang.Integer.parseInt(Integer.java:661)
at org.awaitility.core.JavaVersionDetector.getJavaMajorVersion(JavaVersionDetector.java:21)
at org.awaitility.core.LambdaErrorMessageGenerator.getLambdaDetectionClassName(LambdaErrorMessageGenerator.java:86)
at org.awaitility.core.LambdaErrorMessageGenerator.isLambdaClass(LambdaErrorMessageGenerator.java:30)
at org.awaitility.core.AssertionCondition.generateDescriptionPrefix(AssertionCondition.java:87)
at org.awaitility.core.AssertionCondition.getMatchMessage(AssertionCondition.java:73)
at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:54)
at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248)
at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1447)
...
```

because pre-GA builds of Java have appended `-ea` to the Java version that gets reported back in System.getProperty("java.version"). When that happens, a NumberFormatException is thrown in [JavaVersionDetector](https://github.com/awaitility/awaitility/blob/master/awaitility/src/main/java/org/awaitility/core/JavaVersionDetector.java#L4) because the `-ea` part cannot be turned into a number (for obvious reasons).

Awaitility fixed this in PR https://github.com/awaitility/awaitility/pull/279 and is included in awaitility 4.2.2, now we just need to include the newer version in these two TCKs.

### PR DESCRIPTION
**Pull request by gjwatts**
Fixes #30227 

- [X] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [ ] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by gjwatts** [comment]
#build

**Comment by gjwatts** [comment]
!build

**Review by benjamin-confino — DISMISSED** [review]

**Review by Yas-Franco — DISMISSED** [review]
Looks good

**Comment by Emily-Jiang** [comment]
I think it is better to pull in the MP FT 4.1.1 as it has the update as well. I am going to release MPF 2.0.1 later today. It's better to pull in the version this week rather than fixing our codebase.

**Comment by Emily-Jiang** [comment]
Please pull in MP FT 4.1.1 and MPTelemetry 2.0.1, which has the fix you need.

File: dev/io.openliberty.microprofile.faulttolerance.4.1.internal_fat_tck/publish/tckRunner/tck/pom.xml [hunk 1]
```diff
@@ -65,10 +65,22 @@
...
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <version>4.2.2</version>
+        </dependency>
         <dependency>
             <groupId>org.eclipse.microprofile.fault-tolerance</groupId>
             <artifactId>microprofile-fault-tolerance-tck</artifactId>
             <version>${microprofile.faulttolerance.version}</version>
+                <exclusions>
```
Comment (jhanders34) [inline dev/io.openliberty.microprofile.faulttolerance.4.1.internal_fat_tck/publish/tckRunner/tck/pom.xml]:
```suggestion
              <exclusions>
```

File: dev/io.openliberty.microprofile.telemetry.2.0.internal_fat_tck/publish/tckRunner/tck/pom.xml [hunk 2]
```diff
@@ -52,29 +52,51 @@
...
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <version>4.2.2</version>
+        </dependency>
         <dependency>
             <groupId>org.eclipse.microprofile.telemetry</groupId>
             <artifactId>microprofile-telemetry-tracing-tck</artifactId>
             <version>${mptelemetry.version}</version>
+                <exclusions>
```
Comment (jhanders34) [inline dev/io.openliberty.microprofile.telemetry.2.0.internal_fat_tck/publish/tckRunner/tck/pom.xml]:
```suggestion
              <exclusions>
```

File: dev/io.openliberty.microprofile.telemetry.2.0.internal_fat_tck/publish/tckRunner/tck/pom.xml [hunk 3]
```diff
@@ -52,29 +52,51 @@
...
+                    <groupId>org.awaitility</groupId>
+                    <artifactId>awaitility</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.eclipse.microprofile.telemetry</groupId>
             <artifactId>microprofile-telemetry-metrics-tck</artifactId>
             <version>${mptelemetry.version}</version>
+                <exclusions>
```
Comment (jhanders34) [inline dev/io.openliberty.microprofile.telemetry.2.0.internal_fat_tck/publish/tckRunner/tck/pom.xml]:
```suggestion
              <exclusions>
```

File: dev/io.openliberty.microprofile.telemetry.2.0.internal_fat_tck/publish/tckRunner/tck/pom.xml [hunk 4]
```diff
@@ -52,29 +52,51 @@
...
+                    <groupId>org.awaitility</groupId>
+                    <artifactId>awaitility</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.eclipse.microprofile.telemetry</groupId>
             <artifactId>microprofile-telemetry-logs-tck</artifactId>
             <version>${mptelemetry.version}</version>
+                <exclusions>
```
Comment (jhanders34) [inline dev/io.openliberty.microprofile.telemetry.2.0.internal_fat_tck/publish/tckRunner/tck/pom.xml]:
```suggestion
              <exclusions>
```

**Comment by jhanders34** [comment]
> Please pull in MP FT 4.1.1 and MPTelemetry 2.0.1, which has the fix you need.

Of course those were just published today.  Couldn't move to them last week since they didn't exist.  

@gjwatts when you are back in the office after the holiday, this change gets much easier.

**Review by jhanders34 — CHANGES_REQUESTED** [review]
In light of new versions of the TCK available with the updated awaitility, the change gets much simpler and doesn't require any overriding of dependencies.

**Comment by gjwatts** [comment]
@jhanders34 Thank you for letting me know!

File: dev/io.openliberty.microprofile.faulttolerance.4.1.internal_fat_tck/publish/tckRunner/tck/pom.xml [hunk 5]
```diff
@@ -11,12 +11,12 @@
 
     <groupId>io.openliberty.microprofile.faulttolerance</groupId>
     <artifactId>tck.runner.tck</artifactId>
-    <version>4.1</version>
+    <version>4.1.1</version>
```
Comment (jhanders34) [inline dev/io.openliberty.microprofile.faulttolerance.4.1.internal_fat_tck/publish/tckRunner/tck/pom.xml]: This technically can stay 4.1.  If you look at the 4.0 version of this FAT bucket, it was left to 4.0 for this one.

**Review by jhanders34 — APPROVED** [review]

File: dev/io.openliberty.microprofile.faulttolerance.4.1.internal_fat_tck/publish/tckRunner/tck/pom.xml (hunk 5 above)
Comment (gjwatts) [reply dev/io.openliberty.microprofile.faulttolerance.4.1.internal_fat_tck/publish/tckRunner/tck/pom.xml]: Ok, fixed.  Thanks for letting me know!

**Review by benjamin-confino — APPROVED** [review]

**Review by Yas-Franco — APPROVED** [review]

**Comment by gjwatts** [comment]
!build ([view Open Liberty Personal Build RTC](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=858f3aac-90ad-42f4-9c64-c36d80493075) - :x: completed with errors/failures.) ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=7c9bc719-fd8f-4026-814f-146bc0ada301) - :white_check_mark: completed successfully!)

**Comment by gjwatts** [comment]
Both TCKs have passed multiple times.  Merging.

**Pull request merged**